Add plugin: DocBase (Unofficial)

kuvanov-2 opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["name": "DocBase(Unofficial)",](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/manifest.json#L3)
Add a space between the words

["description": "A simple plugin to pull/push DocBase notes",](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/manifest.json#L6)
Avoid including sentences like "This is a plugin that does ..." in your description, it should be self-evident that this is a plugin for Obsidian.
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

["authorUrl": "https://garden.yuri-k.com/notes/doc-base-unofficial-plugin-for-obsidian/",](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/manifest.json#L8)
This should point to your website, not a website specific to the plugin.

["fundingUrl": "",](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/manifest.json#L9)
This is meant for links to services like _Buy me a coffee_, _GitHub sponsors_ and so on, if you don't have such a link remove this line.

[callback: () => this.pullFromDocBase(),](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/main.ts#L23), [callback: () => this.pushToDocBase(),](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/main.ts#L29)
These should be using [conditional commands](https://docs.obsidian.md/Plugins/User+interface/Commands#Conditional+commands)

[const yamlMatch = content.match(/---\n([\s\S]*?)\n---/);](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/main.ts#L43-L48), [const match = content.match(/docbase_note_id:\s*"(\d+)"|docbase_note_id:\s*(\d+)/);](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/main.ts#L85-L117)
To read the frontmatter use:
```ts
this.app.metadataCache.getFileCache(file)?.frontmatter
```

[const newYaml = `---\ntitle: "${title}"\ndraft: ${draft}\ntags:\n${tags.map((tag: string) => `  - "${tag}"`).join('\n')}\ndocbase_note_id: "${docbaseNoteId}"\n---\n`;](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/main.ts#L61)
use `stringifyYaml` instead of this.

[containerEl.createEl('h2', { text: 'DocBase Plugin Settings' });](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/main.ts#L166)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)


[.setName('Access Token')](https://github.com/kuvanov-2/obsidian-docbase/blob/87670c96191f38fb0b1c32d82dad647ad3383313/main.ts#L169)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

kuvanov-2 commented:
@joethei Thank you for your thorough review. I have made the following modifications based on your feedback:

1. Updated the plugin name in manifest.json to include a space.
2. Revised the plugin description to be more concise and follow the guidelines.
3. Modified the authorUrl to point to my general website.
4. Removed the fundingUrl as it's not applicable.
5. Refactored the commands to use conditional callbacks for improved UX.
6. Updated the frontmatter parsing to use the app.metadataCache.
7. Implemented stringifyYaml for YAML generation instead of manual string construction.
8. Removed the top-level heading from the settings tab.
9. Updated setting names to use sentence case.

You can find these changes in the following commit:
https://github.com/kuvanov-2/obsidian-docbase/commit/28c43169559b7984accb0993efae2b12aa62b763

Please let me know if you need any further modifications or have any questions.
